### PR TITLE
Get binary for specific version of EKS-A release in E2E tests

### PR DESCRIPTION
**Pull request by taneyland**
*Issue #, if available:*

*Description of changes:*
This PR allows us to extract the binary of the desired EKS-A release version for E2E tests.
We also changed the compare function in the semver package to increase its readability.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/650#" title="Author self-approved">taneyland</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [taneyland]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: test/framework/releaseVersions.go
Comment (danbudris) [inline test/framework/releaseVersions.go:90]: consider logging the path as well (here and in the errors)

File: pkg/semver/semver.go
Comment (danbudris) [inline pkg/semver/semver.go:80]: nice this is very clear now

File: test/framework/releaseVersions.go
Comment (danbudris) [inline test/framework/releaseVersions.go:86]: so the directory will be in the format `./bin/v0.6.0/`?

**Review by danbudris — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/650#pullrequestreview-808729038" title="Approved">danbudris</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/650#" title="Author self-approved">taneyland</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [danbudris,taneyland]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: test/framework/releaseVersions.go
Comment (taneyland) [reply test/framework/releaseVersions.go:86]: yes!

File: test/framework/releaseVersions.go
Comment (taneyland) [reply test/framework/releaseVersions.go:48]: Good point. I can change it to `GetLatestMinorReleaseBinaryFromReleaseBranch` because it'll only download the binary if it doesn't exist, but it'll always "get" the binary to use in the E2E test.

**Review by danbudris — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/650#pullrequestreview-808784142" title="Approved">danbudris</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/650#" title="Author self-approved">taneyland</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [danbudris,taneyland]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
